[docs][tools] Docs and scripts for air-gapped registry (#1852)
deckhouse/deckhouse: docs/documentation/pages/DECKHOUSE-FAQ.md

Thus, Deckhouse images will be available at `https://your-harbor.com/d8s/deckhouse/{d8s-edition}:{d8s-version}`.

### Manually upload images to an air-gapped registry

Download script on a host that have access to `registry.deckhouse.io`. `Docker`, `crane` and `jq` must be installed on this host.

```shell
curl -fsSL -o d8-pull.sh https://raw.githubusercontent.com/deckhouse/deckhouse/main/tools/release/d8-pull.sh
chmod 700 d8-pull.sh
./d8-pull.sh --license YOUR_DECKHOUSE_LICENSE_KEY --output-dir /your/output-dir/
```

> Example of pulling images

Reviewer: ```suggestion
Example of pulling images:
```

```shell

> ./d8-push.sh --source-dir /your/source-dir/ --path your.private.registry.com/deckhouse --username YOUR_USERNAME --password YOUR_PASSWORD

Reviewer: ```suggestion
./d8-pull.sh --source-dir /your/source-dir/ --path your.private.registry.com/deckhouse --username YOUR_USERNAME --password YOUR_PASSWORD
```

I'm not sure, but the d8-pull.sh script was downloaded.

Author: You're right, but not only command is wrong. Will fix it in separate commit.

```

> Upload folder from previous step to a host with access to an air-gapped registry. `Crane` must be installed on this host.
> Download script on a host that have access to an air-gapped registry.

Reviewer: ```suggestion
Upload the folder from the previous step to a host with access to an air-gapped registry. `Crane` must be installed on this host.
Download script on a host that has access to an air-gapped registry.
```

```shell
curl -fsSL -o d8-push.sh https://raw.githubusercontent.com/deckhouse/deckhouse/main/tools/release/d8-push.sh
chmod 700 d8-push.sh
.d8-push.sh --source-dir /your/source-dir/ --path your.private.registry.com/deckhouse --username YOUR_USERNAME --password YOUR_PASSWORD
```

> Example of pushing images

Reviewer: ```suggestion
Example of pushing images:
```

```shell
./d8-push.sh --source-dir /your/source-dir/ --path your.private.registry.com/deckhouse --username YOUR_USERNAME --password YOUR_PASSWORD
```

> ## How to bootstrap cluster and run Deckhouse without usage of release channels?

Reviewer: ```suggestion
## How to bootstrap a cluster and run Deckhouse without the usage of release channels?
```

This case is only valid if you don't have release channel images in your air-gapped registry.

> * If you want to bootstrap cluster, you have to use exact tag of a Docker image to install the Deckhouse Platform.

Reviewer: ```suggestion
* If you want to bootstrap a cluster, you have to use the exact tag of a Docker image to install the Deckhouse Platform.
```

For example, if you want to install release v1.32.13, you have to use image `your.private.registry.com/deckhouse/install:v1.32.13`. And you have to use `devBranch: v1.32.13` instead of `releaseChannel: XXX` in `config.yml`.

> * If you already have cluster up and running, you have to remove `releaseChannel` setting from configmap `d8-system/deckhouse` and set appropriate `image` for `d8-system/deckhouse` deployment. Further updates have to be done by manual update of `image`.

Reviewer: ```suggestion
* If you already have a cluster up and running, you have to remove `releaseChannel` setting from the ConfigMap `d8-system/deckhouse` and set the appropriate `image` parameter for `d8-system/deckhouse` Deployment. Further updates have to be done by a manual update of the `image` parameter of the `d8-system/deckhouse` Deployment.
```
